### PR TITLE
Added the functionality for the filepath for SSHUser CheckTrigger

### PR DESCRIPTION
**Pull request by vviveksharma**
### :nut_and_bolt: Description: What code changed, and why?
 Added a functionality in such a way that as the config file sends the file path of key -file it reads the file and add the private to the request body of Batch Check API

### :chains: Related Resources
- https://chefio.atlassian.net/browse/CHEF-2158
### :+1: Definition of Done
- The config should pass the file path to the trigger file and it should be able to read it pass it to SSH User check api
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
- If the key Pair is readable 
Screenshot 2023-06-22 at 8 10 40 PM

- If key reading fails 
Screenshot 2023-06-23 at 11 43 37 AM

### CODE REVIEW

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go [hunk 1]
```diff
@@ -62,14 +67,19 @@ func (ss *SshUserAccessCheck) Run(config *models.Config) []models.CheckTriggerRe
 	return finalResult
 }
 
-func getSShUserAPIRquest(ip string, sshUser *models.SSHUser) models.SShUserRequest {
-
+func (ss *SshUserAccessCheck) getSShUserAPIRquest(ip string, sshUser *models.SSHUser) models.SShUserRequest {
+	ct, err := ss.FileUtils.ReadFile(filepath.Join(os.Getenv("HOME"), sshUser.PrivateKey))
+	if err != nil {
+		ss.log.Error("Error while opeing the file private file present on the path", err)
```
Comment (vivekshankar1) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go:73]: Typo opeing,
Could you modify the line to "Error while opening the file private file present on <actual-path>" ?
Actual-path  is the full path of the file

**Review by vivekshankar1 — APPROVED** [review]

**Review by bvtejaswi — APPROVED** [review]

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go [hunk 2]
```diff
@@ -62,14 +65,20 @@ func (ss *SshUserAccessCheck) Run(config *models.Config) []models.CheckTriggerRe
 	return finalResult
 }
 
-func getSShUserAPIRquest(ip string, sshUser *models.SSHUser) models.SShUserRequest {
-
+func (ss *SshUserAccessCheck) getSShUserAPIRquest(ip string, sshUser *models.SSHUser) models.SShUserRequest {
```
Comment (ArvinthC3000) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go:68]:
```suggestion
func (ss *SshUserAccessCheck) getSShUserAPIRequest(ip string, sshUser *models.SSHUser) models.SShUserRequest {
```

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go [hunk 3]
```diff
@@ -8,20 +8,23 @@ import (
...
 )
 
 type SshUserAccessCheck struct {
-	host string
-	port string
-	log  logger.Logger
+	host      string
+	port      string
+	log       logger.Logger
+	FileUtils fileutils.FileUtils
```
Comment (ArvinthC3000) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go:20]:
```suggestion
	fileUtils fileutils.FileUtils
```

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go [hunk 4]
```diff
@@ -8,20 +8,23 @@ import (
...
-func NewSshUserAccessCheck(log logger.Logger, port string) *SshUserAccessCheck {
+func NewSshUserAccessCheck(log logger.Logger, fileutils fileutils.FileUtils, port string) *SshUserAccessCheck {
 	return &SshUserAccessCheck{
-		log:  log,
-		host: constants.LOCAL_HOST_URL,
-		port: port,
+		log:       log,
+		host:      constants.LOCAL_HOST_URL,
+		port:      port,
+		FileUtils: fileutils,
```
Comment (ArvinthC3000) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go:28]:
```suggestion
		fileUtils: fileutils,
```

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go [hunk 5]
```diff
@@ -44,12 +48,35 @@ func (ss *SshUserAccessCheck) Run(config *models.Config) []models.CheckTriggerRe
...
+		return []models.CheckTriggerResponse{
+			{
+				Status: "SUCCESS",
+				Result: models.ApiResult{
+					Passed:  false,
+					Message: "SSH User Access Check",
+					Check:   "ssh-user",
+					Checks:  []models.Checks{},
+					Error:   &fiber.Error{
+						Code: 400,
```
Comment (ArvinthC3000) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go:63]: Use http.StatusBadRequest or fibererror

File: components/automate-cli/pkg/verifysystemdcreate/systemdcreatesvc.go [hunk 6]
```diff
@@ -28,7 +28,7 @@ ExecStart={{.ServiceCommand}}
 Restart=always
 StandardOutput=journal
 StandardError=journal
-
+Environment="HOME={{.HomeEnv}}"
```
Comment (atultherajput) [inline components/automate-cli/pkg/verifysystemdcreate/systemdcreatesvc.go:31]: Do we need an environment variable as now we are using the absolute path?

File: components/automate-cli/pkg/verifysystemdcreate/mocksystemdcreateutils.go [hunk 7]
```diff
@@ -5,6 +5,7 @@ type MockSystemdCreateUtils struct {
 	SystemdRunningFunc           func() error
 	CreateDestinationAndCopyFunc func(binarySrcPath, binaryDestPath string) error
 	ExecuteShellCommandFunc      func(name string, arg []string) error
+	GetEnvFunc                   func() string
```
Comment (atultherajput) [inline components/automate-cli/pkg/verifysystemdcreate/mocksystemdcreateutils.go:8]: Same as the last comment.

File: components/automate-cli/pkg/verifysystemdcreate/systemdcreatesvc.go [hunk 8]
```diff
@@ -102,12 +103,13 @@ func (css *CreateSystemdServiceImpl) createSystemdServiceFile() error {
 	if err != nil {
 		return errors.Wrap(err, "Error parsing template")
 	}
-
+	envVariable := css.SystemdCreateUtils.GetEnv()
```
Comment (atultherajput) [inline components/automate-cli/pkg/verifysystemdcreate/systemdcreatesvc.go:106]: Same as the last comment.

File: components/automate-cli/pkg/verifysystemdcreate/systemdcreateutils.go [hunk 9]
```diff
@@ -15,6 +15,7 @@ type SystemdCreateUtils interface {
 	SystemdRunning() error
 	CreateDestinationAndCopy(binarySrcPath, binaryDestPath string) error
 	ExecuteShellCommand(name string, arg []string) error
+	GetEnv() string
```
Comment (atultherajput) [inline components/automate-cli/pkg/verifysystemdcreate/systemdcreateutils.go:18]: Same as the last comment.

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go (hunk 4 above)
Comment (vviveksharma) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go]: Done

File: components/automate-cli/pkg/verifysystemdcreate/mocksystemdcreateutils.go (hunk 7 above)
Comment (vviveksharma) [reply components/automate-cli/pkg/verifysystemdcreate/mocksystemdcreateutils.go:8]: Done

File: components/automate-cli/pkg/verifysystemdcreate/systemdcreatesvc.go (hunk 8 above)
Comment (vviveksharma) [reply components/automate-cli/pkg/verifysystemdcreate/systemdcreatesvc.go:106]: Done

File: components/automate-cli/pkg/verifysystemdcreate/systemdcreateutils.go (hunk 9 above)
Comment (vviveksharma) [reply components/automate-cli/pkg/verifysystemdcreate/systemdcreateutils.go:18]: Done

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go (hunk 3 above)
Comment (vviveksharma) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go]: done

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go (hunk 5 above)
Comment (vviveksharma) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go]: done

**Review by ArvinthC3000 — APPROVED** [review]

**Review by vivek-yadav — APPROVED** [review]

**Pull request merged**